Lib/dataclasses.py

```diff
@@ -1267,6 +1268,32 @@ def is_dataclass(obj):
     return hasattr(cls, _FIELDS)
 
 
+class DataclassLike(metaclass=abc.ABCMeta):
+    """Abstract base class for all dataclass types.
+
+    Mainly useful for type-checking.
+    """
+    # __dataclass_fields__ here is really an "abstract class variable",
+    # but there's no good way of expressing that at runtime,
+    # so just make it a regular class variable with a dummy value
+    __dataclass_fields__ = {}
```

Comment:
Why do we need this? In my opinion, `def __subclasshook__(cls, other):` is enough.

For example, your own tests do not use this attribute.

Reply:
But, *is* `DataclassLike` a dataclass? :) 

You can have two opinions here.

Reply:
The contract here is that "all classes for which `is_dataclass` evaluates to `True` shall be considered subclasses of `DataclassLike`. But `DataclassLike` is considered a subclass of `DataclassLike` (because all classes are considered subclasses of themselves in Python). So it would be very strange, in my opinion, for there to be _one_ class (`DataclassLike` itself) where `is_dataclass(x)` evaluates to `False` but it's nonetheless considered a subclass of `DataclassLike`.

Reply:
Alternatively, I could make `DataclassLike` an _actual_ dataclass, I suppose?